errbot/plugin_wizard.py

```diff
@@ -0,0 +1,144 @@
+
+from errbot import PY2
+
+if PY2:
+    input = raw_input
+
+
+def new_plugin_wizard(directory=None):
+    """
+    Start the wizard to create a new plugin in the current working directory.
+    """
+    if directory is None:
+        print("This wizard will create a new plugin for you in the current directory.")
+        directory = os.getcwd()
+    else:
+        print("This wizard will create a new plugin for you in '%s'." % directory)
+
+    if not os.path.isdir(directory):
+        print("Error: The path '%s' does not exist or is not a directory." % directory)
+        sys.exit(1)
+
+    try:
+        plugin_code = download_plugin_skeleton()
+    except requests.HTTPError as e:
+        print(
+            "Error: I couldn't download the plugin skeleton from GitHub "
+            "which is needed to write out a new plugin file for you."
+        )
+        print("Details: {!s}".format(e))
+        sys.exit(1)
+
+    name = ask(
+        "What should the name of your new plugin be?",
+        validation_regex=r'^[a-zA-Z][a-zA-Z0-9 _-]*$'
+    ).strip()
+    module_name = name.lower().replace(' ', '_')
+    class_name = "".join([s.capitalize() for s in name.lower().split(' ')])
+
+    description = ask(
+        "What may I use as a short (one-line) description of your plugin?"
+    )
+    python_version = ask(
+        "Which python version will your plugin work with? 2, 2+ or 3?",
+        valid_responses=['2', '2+', '3']
+    )
+    errbot_min_version = ask(
+        "Which minimum version of errbot will your plugin work with? Leave blank to support any version"
+    ).strip()
+    errbot_max_version = ask(
+        "Which maximum version of errbot will your plugin work with? Leave blank to support any version"
+    ).strip()
+
+    plug = configparser.ConfigParser()
+    plug["Core"] = {
+        "Name": name,
+        "Module": module_name,
+    }
+    plug["Documentation"] = {
+        "Description": description,
+    }
+    plug["Python"] = {
+        "Version": python_version,
+    }
+    plug["Errbot"] = {}
+    if errbot_min_version != "":
+        plug["Errbot"]["Min"] = errbot_min_version
+    if errbot_max_version != "":
+        plug["Errbot"]["Max"] = errbot_max_version
+
+    plug_path = os.path.join(directory, module_name+".plug")
+    py_path = os.path.join(directory, module_name+".py")
+    if os.path.exists(plug_path) or os.path.exists(py_path):
+        print(
+            "Warning: A plugin with this name was already found at {path}\n"
+            "If you continue, these will be overwritten.".format(
+                path=os.path.join(directory, module_name+".{py,plug}")
+            )
+        )
+        ask(
+            "Press Ctrl+C to abort now or type in 'overwrite' to confirm overwriting of these files.",
+            valid_responses=["overwrite"],
+        )
+
+    with open(plug_path, 'w') as f:
+        plug.write(f)
+
+    plugin_code = plugin_code.replace("class Skeleton(", "class %s(" % class_name)
```

Comment:
I think I might prefer maybe a simpler version of the skeleton plugin bundled with errbot so we can maintain it consistently with errbot. This looks very fragile if in another repo IMHO.
Then the Jinja markers are not really a problem as they are internal details.